consensus/src/aggregation/config.rs

```diff
@@ -0,0 +1,98 @@
+use super::types::{Activity, Epoch, Index};
+use crate::{Automaton, Monitor, Reporter, ThresholdSupervisor};
+use commonware_cryptography::{bls12381::primitives::variant::Variant, Digest};
+use commonware_utils::Array;
+use std::time::Duration;
+
+/// Configuration for the [`Engine`](super::Engine).
+pub struct Config<
+    P: Array,
+    V: Variant,
+    D: Digest,
+    A: Automaton<Context = Index, Digest = D>,
+    Z: Reporter<Activity = Activity<V, D>>,
+    M: Monitor<Index = Epoch>,
+    TSu: ThresholdSupervisor<Index = Epoch, PublicKey = P>,
+> {
+    /// Tracks the current state of consensus (to determine which participants should
+    /// be involved in the current broadcast attempt).
+    pub monitor: M,
+
+    /// Manages the set of validators and the group identity.
+    /// Also manages the cryptographic partial share if the engine is a validator.
+    pub validators: TSu,
+
+    /// Proposes and verifies digests.
+    pub automaton: A,
+
+    /// Notified when a chunk receives a threshold of acks.
+    pub reporter: Z,
+
+    /// The application namespace used to sign over different types of messages.
+    /// Used to prevent replay attacks on other applications.
+    pub namespace: Vec<u8>,
+
+    /// Whether acks are sent as priority.
+    pub priority_acks: bool,
+
+    /// How often an ack is rebroadcast to all validators if no threshold is reached.
+    pub rebroadcast_timeout: Duration,
+
+    /// A tuple representing the epochs to keep in memory.
+    /// The first element is the number of old epochs to keep.
+    /// The second element is the number of future epochs to accept.
+    ///
+    /// For example, if the current epoch is 10, and the bounds are (1, 2), then
+    /// epochs 9, 10, 11, and 12 are kept (and accepted);
+    /// all others are pruned or rejected.
+    pub epoch_bounds: (u64, u64),
+
+    /// The concurrent number of chunks to process.
+    pub window: u64,
+
+    /// A unique name for the journal.
+    pub journal_name: String,
```

Comment:
nit: in other areas, we call this `partition`: https://github.com/commonwarexyz/monorepo/blob/bf35da20100d131320b77d0a8b70428e5799631b/consensus/src/threshold_simplex/config.rs#L48-L49

Reply:
https://github.com/commonwarexyz/monorepo/pull/974/commits/e96fb501d1c8dab5edff7ca6e0df00fa40d5e6fb